Modules/System/SharePoint/src/SharePointClientImpl.Codeunit.al

```diff
@@ -494,6 +494,22 @@ codeunit 9101 "SharePoint Client Impl."
         exit(true);
     end;
 
+    procedure DeleteFile(OdataId: Text): Boolean
+    var
+        SharePointHttpContent: Codeunit "SharePoint Http Content";
+    begin
+        //POST https://{site_url}/_api/web/GetFileByServerRelativeUrl('/Folder Name/{file_name}')
```

Comment:
@jwikman I don't quite understand this approach to delete a file.
I know it's documented on the following page: https://learn.microsoft.com/en-us/sharepoint/dev/sp-add-ins/working-with-folders-and-files-with-rest

But isn't it possible to delte a sharepoint list item by using the graph API:
If it is possible then this restful API call should be added as option as well in my opinion.
https://learn.microsoft.com/en-us/graph/api/listitem-delete?view=graph-rest-1.0&tabs=http

If I understand the current code correctly, then they make use of both API right now..?
Or is it the same API?

Reply:
@pri-kise @jwikman 

For your information, this request also works perfectly to delete a file:
//DELETE https://{site_url}/_api/web/GetFileByServerRelativeUrl('/Folder Name/{file_name}')

It may be better to implement this method rather than using POST with the XHttpMethod header.

Reply:
@jwikman 

There is no example of this kind of code in the documentation. But it is clearly stated that we can use these REST endpoints to perform typical CRUD operations on SharePoint entities.

documentation here : 
- https://learn.microsoft.com/en-us/sharepoint/dev/sp-add-ins/get-to-know-the-sharepoint-rest-service?tabs=csom

<html>
<body>
<!--StartFragment--><div class="heading-wrapper" data-heading-level="h3" style="box-sizing: inherit; outline-color: inherit; position: relative; color: rgb(230, 230, 230); font-family: &quot;Segoe UI&quot;, SegoeUI, &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-serif; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(23, 23, 23); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h3 id="use-http-commands-with-the-sharepoint-rest-service" class="heading-anchor" style="box-sizing: inherit; outline-color: inherit; margin: 30px 0px 18px; padding: 0px; font-size: clamp(1.3125rem, 15.4737px + 1.15132vw, 1.75rem); font-weight: 600; line-height: 1.3; margin-inline-start: -1.875rem; padding-inline-start: 1.875rem;">Use HTTP commands with the SharePoint REST service</h3></div><p style="box-sizing: inherit; outline-color: inherit; margin: 1rem 0px 0px; padding: 0px; overflow-wrap: break-word; color: rgb(230, 230, 230); font-family: &quot;Segoe UI&quot;, SegoeUI, &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-serif; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(23, 23, 23); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">To use the REST capabilities that are built into SharePoint, you construct a RESTful HTTP request by using the OData standard, which corresponds to the client object model API you want to use. The client.svc web service handles the HTTP request and serves the appropriate response in either Atom or JSON format. The client application must then parse that response.</p><p style="box-sizing: inherit; outline-color: inherit; margin: 1rem 0px 0px; padding: 0px; overflow-wrap: break-word; color: rgb(230, 230, 230); font-family: &quot;Segoe UI&quot;, SegoeUI, &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-serif; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(23, 23, 23); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">The endpoints in the SharePoint REST service correspond to the types and members in the SharePoint client object models. By using HTTP requests, you can use these REST endpoints to do typical CRUD operations against SharePoint entities, such as lists and sites.</p><div class="has-inner-focus" style="box-sizing: inherit; outline-color: inherit; color: rgb(230, 230, 230); font-family: &quot;Segoe UI&quot;, SegoeUI, &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-serif; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(23, 23, 23); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">

If you want to do this to an endpoint | Use this HTTP request | Keep in mind
-- | -- | --
Read a resource | GET |  
Create or update a resource | POST | Use POST to create entities such as lists and sites.The SharePoint REST service supports sending POST commands that include object definitions to endpoints that represent collections.For POST operations, any properties that aren't required are set to their default values.If you attempt to set a read-only property as part of a POST operation, the service returns an exception.
Update or insert a resource | PUT | Use PUT and MERGE operations to update existing SharePoint objects.Any service endpoint that represents an object property set operation supports both PUT requests and MERGE requests.For MERGE requests, setting properties is optional; any properties that you don't explicitly set keep their current property.For PUT requests, if you don't specify all required properties in object updates, the REST service returns an exception.In addition, any optional properties you don't explicitly specify are set to their default properties.
Delete a resource | DELETE | Use the HTTP DELETE command against the specific endpoint URL to delete the SharePoint object represented by that endpoint.There are recyclable objects, such as lists, files, and list items, this results in a Recycle operation.

</div><!--EndFragment-->
</body>
</html>

Reply:
Here is my fork with the addition of the Delete Http method in SharePoint Request Helper.
It also adds SharePointClient.DeleteFile(), SharePointClient.DeleteFolder() and SharePoint.DownloadFileContent() overloads based on your PR. 

https://github.com/glachana/ALAppExtensions/tree/dev/sharepoint-upgrade-client 

Let me know if a PR from my branch might be worth it ;)

Reply:
Thanks for clarifying @glachana 

I don't like the X-Http-Method approach, so as long as the regular DELETE approach works - I think it is better.
Why do they even have the X-Http-Method implementation if DELETE works...? 🤔

I don't mind if this PR is scraped in advantage for yours with DELETE method.

Maybe someone else has an opinion on this? @dynaway-kmi or @pri-kise?

Reply:
Here is my PR #22988. It contains the changes included in these two PR #22565 #2240. Regarding the delete function, it uses DELETE HTTP Commands instead of X-Http-Method implementation.